File: 9 - Recursion/Solution.java
import java.util.Scanner;

public class Solution {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int n = in.nextInt();
        in.close();
        System.out.println(factorial(n));
    }

    private static int factorial(int n) {
        if (n == 1) return 1;
        return factorial(n - 1) * n;
    }
}

File: 9 - Recursion/Solution.cpp
#include <iostream>

using namespace std;

int factorial(int n) {
    if (n == 1) return 1;
    return factorial(n - 1) * n;
}

int main() {
    int N;
    cin >> N;

    cout << factorial(N);

    return 0;
}
